demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,511 @@
+                    classIndices.push_back(static_cast<unsigned>(std::stoul(imageSearchResult->second)));
+                } else {
+                    throw std::runtime_error("No class specified for image " + imageNames[i]);
+                }
+            }
+        } else {
+            classIndices.resize(inputImages.size());
+            std::fill(classIndices.begin(), classIndices.end(), 0);
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // --------------------------------------------Read labels--------------------------------------------
+        std::vector<std::string> labels;
+        std::ifstream inputLabelsFile(FLAGS_labels);
+        std::string labelsLine;
+        while (std::getline(inputLabelsFile, labelsLine)) {
+            labels.push_back(labelsLine.substr(labelsLine.find(' ') + 1,
+                                               labelsLine.find(',') - (labelsLine.find(' ') + 1)));
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // -------------------------------------------Read network--------------------------------------------
+        Core ie;
+        CNNNetwork network = ie.ReadNetwork(FLAGS_m);
+        // ---------------------------------------------------------------------------------------------------
+
+        // ------------------------------------------Reshape network------------------------------------------
+        auto inputShapes = network.getInputShapes();
+        std::string inputName;
+        SizeVector inputShape;
+        std::tie(inputName, inputShape) = *inputShapes.begin();
+        if (inputShapes[inputName].size() != 4) {
+            throw std::logic_error("Model input has incorrect number of dimensions. Must be 4.");
+        }
+        if (inputShapes[inputName][1] != 3) {
+            throw std::logic_error("Model input has incorrect number of color channels."
+                                   " Expected 3, got " + std::to_string(inputShapes[inputName][1]) + ".");
+        }
+        if (inputShapes[inputName][2] != inputShapes[inputName][3]) {
+            throw std::logic_error("Model input has incorrect image shape. Must be NxN square."
+                                   " Got " + std::to_string(inputShapes[inputName][2]) + 
+                                   "x" + std::to_string(inputShapes[inputName][3]) + ".");
+        }
+        int modelInputResolution = inputShapes[inputName][2];
+        inputShape[0] = FLAGS_b;
+        inputShape[2] = inputShapes[inputName][2];
+        inputShape[3] = inputShapes[inputName][3];
+        inputShapes[inputName] = inputShape;
+        std::cout << "Resizing network to the image size = [" 
+                  << inputShapes[inputName][2] << "x" << inputShapes[inputName][3]
+                  << "] " << "with batch = " << FLAGS_b << std::endl;
```

Comment:
The message should be removed: that's not image size. The model is reshaped to the same width and height as it was. The part about batch is correct, but I doubt it is useful: the demo just does what it was asked to do.